pkgs/by-name/ls/lsp-ai/package.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  cmake,
+  openssl,
+  zlib,
+  perl,
+}:
+
+let
+  pname = "lsp-ai";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "SilasMarvin";
+    repo = pname;
+    rev = "66655f6285fe8aad8f2d72646e9ed47c06245525";
+    hash = "sha256-DwqqZBzLevuRCW6QzGyWdE+JtpW6b3EMDuiWtajv/U4=";
+  };
+
+in
+rustPlatform.buildRustPackage {
+  inherit pname version src;
```

Comment:
```suggestion
rustPlatform.buildRustPackage {
  pname = "lsp-ai";
  version = "0.7.0";

  src = fetchFromGitHub {
    owner = "SilasMarvin";
    repo = "lsp-ai";
    rev = "66655f6285fe8aad8f2d72646e9ed47c06245525";
    hash = "sha256-DwqqZBzLevuRCW6QzGyWdE+JtpW6b3EMDuiWtajv/U4=";
  };
```

Reply:
Is this because we don't want too much stuff to be pre-processed before starting a build?

Reply:
> Is this because we don't want too much stuff to be pre-processed before starting a build?

It's the same putting outside of build function and inside. Nix evaluates them in the same way, computes the derivation (`.drv`) and starts the build.

This is corrected because `repo = pname` pattern isn't useful and it is unnecessary to put several attributes into let-in to create condition for it.

Reply:
Gutted the entire let statement. Much cleaner now.